### PR TITLE
Fix exec for Windows

### PR DESCRIPTION
**Pull request by genotrance**

### CODE REVIEW

File: issue280and524.nimble [hunk 1]
```diff
@@ -13,7 +13,8 @@ requires "nim >= 0.18.0"
 task setup, "Download and generate":
   exec "nimble install https://github.com/nimble-test/issue280and524.git?subdir=generator -y"
   withDir thisDir():
-    exec "generator"
+    let cmd = when defined(windows): "cmd /c " else: ""
+    exec cmd & "generator"
```
Comment (dom96) [inline issue280and524.nimble:17]: Wouldn't ``addFileExt(ExeExt)`` work too? (I would prefer it)

**Comment by genotrance** [comment]
There is no exe in the path, it is a.cmd file on Windows that cannot run without cmd.exe.

**Comment by dom96** [comment]
Why doesn't `exec` run it with cmd.exe already?

**Comment by genotrance** [comment]
[exec](https://github.com/nim-lang/Nim/blob/devel/lib/system/nimscript.nim#L238) calls [rawExec](https://github.com/nim-lang/Nim/blob/devel/compiler/scriptconfig.nim#L90) which calls [execCmd](https://github.com/nim-lang/Nim/blob/devel/lib/pure/osproc.nim#L743) which is calling CreateProcess directly on Windows. There's no cmd in the picture since csystem() isn't being used. Also, nimble creates .cmd shortcuts on Windows for binaries which require cmd to run.

**Pull request merged**